src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -830,12 +813,8 @@ - (BOOL)_subscriptionAbleToReport
 #endif
 
     // Unfortunately, we currently have no subscriptions over our hacked-up XPC
-    // setup.  Try to detect that situation.
-    if ([_deviceController isKindOfClass:MTRDeviceControllerOverXPC.class]) {
-        return NO;
-    }
-
-    return YES;
+    // setup. Based on whether subscriptions are allowed or not, return YES or NO.
```

Comment:
How about:
```suggestion
    // Subscriptions are not able to report if they are not allowed.
```

and then the XPC bits are just an impl detail of `_subscriptionsAllowed`.